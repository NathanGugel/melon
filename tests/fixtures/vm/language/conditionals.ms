let a = 5;
let b = 6;

if (b > a)
    io.print("b > a 1");

if (b > a) {
    io.print("b > a 2");
}

if (!(a > b)) {
    io.print("!(a > b)");
}

if (a > b) { }
else io.print("a < b");

if (a == 6) {}
else if (a == 7) {}
else if (a == 5) {
    io.print("a == 5");
}

if (a > 0) {
    if (a < 7) {
        if ((a == 5) == true) {
            io.print("a == 5 2");
        }
    }
}

if (a >= 5 && b <= 6 && (a > 7 || b < 7)) {
    io.print("a >= 5 && b <= 6 && (a > 7 || b < 7)");
}

if (a > 10 && b < 6 || true) {
    io.print("a > 10 && b < 6 || true");
}