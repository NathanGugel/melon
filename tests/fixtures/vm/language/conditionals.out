b > a 1
b > a 2
!(a > b)
a < b
a == 5
a == 5 2
a >= 5 && b <= 6 && (a > 7 || b < 7)
a > 10 && b < 6 || true
